src/components/TestResult/TestResult.test.tsx

```diff
@@ -0,0 +1,208 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+import { fireEvent, waitFor } from '@testing-library/react';
+import React from 'react';
+import { createStep } from '../../../common/helper/test/createAction';
+import { render } from '../../helpers/test';
+import { TestResult } from './TestResult';
+
+describe('<TestResult />', () => {
+  it('does not render the flyout if there is no result data', () => {
```

Comment:
I'm wondering if it makes sense to improve this by showing _Waiting for tests result.._ or something similar rather than not showing it at all. But this is just an idea and out of the scope of this PR

Reply:
Hm, if testing is in progress and we're awaiting the result, that could be a better UX. I'll add a ticket to address that post v1.

Reply:
https://github.com/elastic/synthetics-recorder/issues/365